src/components/side_panel/chart/building_blocks/generic_side_panel/config_panel.ts

```diff
@@ -109,6 +155,89 @@ export class GenericChartConfigPanel extends Component<Props, SpreadsheetChildEn
     });
   }
 
+  get datasetStructures() {
+    return [
+      { value: "AsColumns", label: _t("As Columns") },
+      { value: "AsRows", label: _t("As Rows") },
+    ];
+  }
+
+  setDatasetStructure(value: string) {
```

Comment:
```suggestion
  setDatasetStructure(value: "AsRows" | "AsColumns") {
```
should probably be a global type